### PR TITLE
Marketing Survey: Add survey for improved Jetpack cancellation flow

### PR DESCRIPTION
**Pull request by IanRamosC**
#### Changes proposed in this Pull Request

* This component adds the survey that will be used as a step of the new Jetpack cancellation flow via Calypso.
* This PR is related to/ a sub-part of https://github.com/Automattic/wp-calypso/pull/54669 which introduces a new cancellation flow for Jetpack plans/ products

#### Testing instructions

The visual output of this component can be seen by testing PR #54669
* Either check out the branch from PR #54669 locally or use the Calypso live link on that branch to get a preview. 
* Review the testing instructions on PR #54669 to check the visual output
* Make sure it looks like the screenshot below

image

### CODE REVIEW

File: client/components/marketing-survey/cancel-jetpack-form/jetpack-cancellation-survey.tsx [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+				key={ choiceOther.id }
+			>
+				<div className="jetpack-cancellation-survey__card-content">
+					<span>{ choiceOther.answerText }</span>
+					<FormTextInput
+						inputRef={ customAnswerInputRef }
+						className="jetpack-cancellation-survey__choice-item-text-input"
+						value={ customAnswerText }
+						onChange={ onChangeCustomAnswerText }
+						placeholder={ translate( 'share your experience' ) }
```
Comment (a8ci18n) [inline client/components/marketing-survey/cancel-jetpack-form/jetpack-cancellation-survey.tsx:120]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **27** times:
`translate( 'share your experience…' )` *ES Score: 9*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/survey-for-improved-jetpack-cancellation-flow)

File: client/components/marketing-survey/cancel-jetpack-form/jetpack-cancellation-survey.tsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	};
+
+	const selectAnswer = useCallback(
+		( answerId: string, answerText: TranslateResult | string ) => {
+			// prevent from changing the answer text to choiceOther.answerText just by clicking on it
+			const surveyAnswerText =
+				choiceOther.id === answerId && customAnswerText ? customAnswerText : answerText;
+
+			setSelectedAnswerId( answerId );
+			onAnswerChange( answerId, surveyAnswerText );
```
Comment (jboland88) [inline client/components/marketing-survey/cancel-jetpack-form/jetpack-cancellation-survey.tsx:58]: I think the way that the surveys are recorded, the "text" field is intended for only a custom typed response, here the question text is passed back to the parent component unless it is the `customAnswerText`. Maybe an empty string should be passed if the custom answer is not used.
This seems to be the case in the product cancellation now in `cancel-purchase-form/index.js`: https://github.com/Automattic/wp-calypso/blob/b294868a28ac029094e06c9b9714b0b34a057db9/client/components/marketing-survey/cancel-purchase-form/index.jsx#L171-L190

File: client/components/marketing-survey/cancel-jetpack-form/jetpack-cancellation-survey.tsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+interface JetpackCancellationSurveyProps {
+	onAnswerChange: ( answerId: string | null, answerText: TranslateResult | string ) => void;
+}
+
+export default function JetpackCancellationSurvey( {
+	onAnswerChange,
+}: JetpackCancellationSurveyProps ): ReactElement {
+	const translate = useTranslate();
+	const [ selectedAnswerId, setSelectedAnswerId ] = useState< string | null >( null );
+	const [ customAnswerText, setCustomAnswerText ] = useState< string >( '' );
```
Comment (jboland88) [inline client/components/marketing-survey/cancel-jetpack-form/jetpack-cancellation-survey.tsx]: Having the selected answer ID and text here as well as in the state of the parent component may be redundant - would it make sense to pass these values as props?

File: client/components/marketing-survey/cancel-jetpack-form/jetpack-cancellation-survey.tsx [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+			{ /* The card for the 'other' option */ }
+			<Card
+				className={ classnames( 'jetpack-cancellation-survey__card', {
+					'is-selected': choiceOther.id === selectedAnswerId,
+				} ) }
+				tagName="button"
+				onClick={ () => {
+					selectAnswer( choiceOther.id, choiceOther.answerText );
+					customAnswerInputRef?.current?.focus();
```
Comment (jboland88) [inline client/components/marketing-survey/cancel-jetpack-form/jetpack-cancellation-survey.tsx:110]: Nice - I appreciate the auto-focus of the field 👍

File: client/components/marketing-survey/cancel-jetpack-form/style.scss [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+			padding: 1rem 4rem;
+		}
+
+		@include break-wide {
+			width: 1128px;
+			padding: 3rem 12rem;
+		}
+	}
+
+	&__card {
```
Comment (jboland88) [inline client/components/marketing-survey/cancel-jetpack-form/style.scss:49]: It looks like we need to make the styling for the first and second steps in the cancellation flow more visually consistent. The modal padding, card size and font sizes look to be different between the two steps. Some classes like `.cancel-jetpack-form__step-headline` that are already defined here can be used - maybe defining other styles for all steps in this file would be a better pattern to bring consistency.

File: client/components/marketing-survey/cancel-jetpack-form/jetpack-cancellation-survey.tsx [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+					<span>{ choice.answerText }</span>
+				</div>
+				<Gridicon icon="chevron-right" size={ 12 } />
+			</Card>
+		);
+	};
+
+	return (
+		<div className="jetpack-cancellation-survey__content">
+			<FormattedHeader
```
Comment (jboland88) [inline client/components/marketing-survey/cancel-jetpack-form/jetpack-cancellation-survey.tsx:93]: What do you think about using `.cancel-jetpack-form__step-headline` here to have the headlines look the same between all steps in the cancellation flow?

**Review by jboland88 — CHANGES_REQUESTED** [review]
A few changes outlined in the comments :)
I also noticed that the first and second steps in the cancellation flow look visually different - I think it may be best to define some of the styles that will apply to all the steps in the `cancel-jetpack-form/style.scss` file - some of this styling may be something that is easier done after the smaller sub-parts of this feature are already merged and we are preparing the final PR to enable the new cancellation flow since styling is broken up across different PRs at the moment.

**Review by jboland88 — APPROVED** [review]
This LGTM 👍  Thanks for the revisions!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6471374

Thank you @IanRamosC for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6471374).

**Pull request merged**